src/components/pagination/pagination.jsx

```diff
@@ -0,0 +1,199 @@
+import { createContext, useContext, forwardRef } from 'react';
+import { cn } from '@/utilities/functions';
+import { ChevronLeft, ChevronRight } from 'lucide-react';
+import { disabledClassNames, sizeClassNames } from './component-style';
+import Button from '../button';
+
+const PageContext = createContext();
+const usePageContext = () => useContext( PageContext );
+
+const Pagination = ( {
+	size = 'sm',
+	disabled = false,
+	children,
+	className,
+	...props
+} ) => (
+	<PageContext.Provider value={ { size, disabled } }>
+		<nav
+			role="navigation"
+			aria-label="pagination"
+			className={ cn(
+				'flex w-full justify-center box-border m-0',
+				className
+			) }
+			{ ...props }
+		>
+			{ children }
+		</nav>
+	</PageContext.Provider>
+);
+Pagination.displayName = 'Pagination';
+
+const PaginationContent = forwardRef( ( { className, ...props }, ref ) => {
+	return (
+		<ul
+			ref={ ref }
+			className={ cn(
+				'm-0 p-0 w-full flex justify-center flex-row items-center gap-1',
+				'list-none',
+				className
+			) }
+			{ ...props }
+		/>
+	);
+} );
+PaginationContent.displayName = 'Pagination.Content';
+
+const PaginationItem = forwardRef(
+	( { isActive = false, href = '', className, children, ...props }, ref ) => {
+		const { disabled } = usePageContext();
+		return (
+			<li
+				ref={ ref }
+				tabIndex={ 0 }
+				className={ cn( 'flex', disabled && disabledClassNames.general ) }
+			>
+				<PaginationButton
+					isActive={ isActive }
+					disabled={ disabled }
+					className={ className }
+					href={ href }
+					{ ...props }
+				>
+					{ children }
+				</PaginationButton>
+			</li>
+		);
+	}
+);
+PaginationItem.displayName = 'Pagination.Item';
+
+const PaginationButton = ( {
+	icon = null,
+	isActive = false,
+	href = '',
+	target = '_self',
+	rel,
+	children,
+	className,
+	...props
+} ) => {
+	const { size, disabled } = usePageContext();
+
+	return (
+		<Button
+			tag="a"
```

Comment:
Instead of setting it statically, we can leave this to the user, and we can set the tag value to a by default if not passed by users. Also, parameters related to the `a` tag should not be set by default. If a developer wants to use React Router Link as the tag value, in that case it will create an issue.